be-python-flask/Jenkinsfile

```diff
@@ -1,77 +1,18 @@
-// build params
-def odsImageTag = env.ODS_IMAGE_TAG ?: 'latest'
 def odsGitRef = env.ODS_GIT_REF ?: 'production'
-def projectId = env.PROJECT_ID
-def componentId = env.COMPONENT_ID.toLowerCase()
-def packageName = env.PACKAGE_NAME
-def gitUrlHttp = env.GIT_URL_HTTP
-def group = env.GROUP_ID
+def odsImageTag = env.ODS_IMAGE_TAG ?: 'latest'
+library("ods-jenkins-shared-library@${odsGitRef}")
 
-def credentialsId = "${projectId}-cd-cd-user-with-password"
-def quickstarterBasepath = "be-python-flask"
-def dockerRegistry
-def gitHost
-// vars from jenkins master
-node {
-  dockerRegistry = env.DOCKER_REGISTRY
-  gitHost =  env.BITBUCKET_HOST.split(":")[0]
-}
+odsQuickstarterPipeline(
+  imageStreamTag: "cd/jenkins-slave-python:${odsImageTag}",
+) { context ->
+  odsQuickstarterStageCopyFiles(context)
 
-def conts = containerTemplate(
-  name: 'jnlp',
-  image: "${dockerRegistry}/cd/jenkins-slave-python:${odsImageTag}",
-  workingDir: '/tmp',
-  alwaysPullImage: true,
-  args: ''
-)
+  odsQuickstarterStageCreateOpenShiftResources(
+    context,
+    [directory: 'common/ocp-config/component-environment']
+  )
 
-def podLabel = "quickstarter-${quickstarterBasepath}-${projectId}-${componentId}"
+  odsQuickstarterStageRenderJenkinsfile(context)
 
-podTemplate(
-      label: podLabel,
-      cloud: 'openshift',
-      containers: [conts],
-      volumes: [],
-      serviceAccount: 'jenkins'
-    ) {
-        node(podLabel) {
-          stage('Checkout quickstarter') {
-            checkout scm
-          }
-          stage("initialize target folder ${componentId}") {
-              sh "mkdir -p ${componentId}"
-          }
-          stage('copy files from quickstart to generated project') {
-              sh "cp -rv ${quickstarterBasepath}/README.md ${quickstarterBasepath}/files"
```

Comment:
approved!

just one question: why are not we providing an initial readme file to the provisioned quickstarters as done here for example?

Reply:
hmm ... I guess up to the quickstarter owner? In general, this is probably a good idea :)

Reply:
shall I make a proposal fix on the ODS lib side then? so to have also a cp of the readme into the files/target folder?

Reply:
@gerardcl I'd rather leave this to the quickstarter author. If they place a `README.md` into `files`, great. Otherwise, it will not be there. I think a `README.md` in the "quickstarter directory" can (and should?) serve a different purpose.